### ISSUE TITLE
Bug: RSI calculation — division by zero when all losses = 0 (flat market)

### ISSUE THREAD

**Issue by artfhc** [artfhc/clawdbot-spy-signals#1]
## Description

When the market has no down days over the 14-day RSI window, `loss` is `0` and `rs = gain / loss` produces `inf` or `NaN`, propagating invalid RSI values silently.

## Affected Files

| File | Line |
|------|------|
| `daily_signal_checker.py` | 30 |
| `signal_alerts.py` | 35, 46 |
| `alpaca_trader.py` | 46 |
| `dashboard.py` | 45, 50 |

## Repro

```python
# Flat market: all deltas = 0, gain = 0, loss = 0
rs = gain / loss  # ZeroDivisionError or inf/NaN
data['RSI'] = 100 - (100 / (1 + rs))  # produces NaN
```

## Fix

Replace the bare division with a safe form:

```python
loss = loss.replace(0, 1e-10)  # prevent division by zero
rs = gain / loss
data['RSI'] = 100 - (100 / (1 + rs))
```

Or use `np.where`:
```python
data['RSI'] = np.where(loss == 0, 100, 100 - (100 / (1 + gain / loss)))
```

This same fix must be applied in all four files.
